### PR TITLE
Fix Issue 15094 - __traits(getMember) fails when the source is a struct/class field

### PR DESCRIPTION
**Pull request by JinShil**
Revival of #5215
Special thanks to @9rnsr

### CODE REVIEW

File: src/ddmd/traits.d [hunk 1]
```diff
@@ -58,6 +59,31 @@ struct PushAttributes
...
+ * additionally strip off expression contexts.
+ *
+ * Some symbol related `__traits` ignore arguments expression contexts.
+ * For example:
+ *  struct S { void f() {} }
+ *  S s;
+ *  pragma(msg, __traits(isNested, s.f));
+ *  // s.f is DotVarExp, but __traits(isNested) needs a FuncDeclaration.
+ *
+ * This is used for that common `__traits` behavior.
```
Comment (WalterBright) [inline src/ddmd/traits.d]: Function comment is not following Ddoc style.

Comment (JinShil) [reply src/ddmd/traits.d]: Done.  Thanks!

**Review by andralex — APPROVED** [review]
awesome possum

**Comment by andralex** [comment]
I'll pull this, looks soild. cc @WalterBright

**Pull request merged**